src/core/models/model_list.py

```diff
@@ -3,15 +3,80 @@
 from core.models.model import ActuatorModel, EnvironmentModel, SensorModel
 from core.state import State, array_to_state
 from core.config import Config
-from utils.constants import ModelEnum, State_Type
+from utils.constants import BodyEnum, ModelEnum, State_Type
+from utils.astropy_util import get_ephemeris
+from astropy.constants import G, M_earth, M_sun
 
 
 class AttitudeDynamics(EnvironmentModel):
     ...
 
 
 class PositionDynamics(EnvironmentModel):
-    ...
+    """
+    The position dynamics model implementation.
+    """
+
+    def __init__(self, parameters) -> None:
+        super().__init__(parameters)
+
+    def evaluate(self, t: float, state: State) -> Dict[str, State_Type]:
+        return super().evaluate(t, state)
+
+    def d_state(self, t: float, state: State) -> Dict[str, State_Type]:
+        """
+        Takes the derivative of a vector [r v] to compute [v a], where r is a position vector,
+        v is the velocity vector, and a is the acceleration vector
+        Args:
+            t (float): the initial time
+            state (State): the initial state
+
+        Returns:
+            Dict[str, State_Type]: The updated vector [v a]
+        """
+        # position column vectors from moon/sun/earth/craft to the origin, where the origin is the Earth's center of mass
+
+        # craft to origin
+        r_co = np.array([state.x, state.y, state.z])
+        # moon to origin
+        r_mo = np.array(get_ephemeris(t, BodyEnum.Moon))
+        # sun to origin
+        r_so = np.array(get_ephemeris(t, BodyEnum.Sun))
+        # earth to origin
+        r_eo = np.array((0.0, 0.0, 0.0))  # Earth is at the origin in GCRS
+
+        # position column vectors from body to the craft
+
+        # moon to the craft
+        r_mc = np.subtract(r_mo, r_co)
+        # sun to the craft
+        r_sc = np.subtract(r_so, r_co)
+        # earth to the craft
+        r_ec = np.subtract(r_eo, r_co)
+
+        # mu values of the body, where mu = G * m_body
+        mu_moon = G * 7.34767309e22
+        mu_sun = G * 1.988409870698051e30
+        mu_earth = G * 5.972167867791379e24
+
+        # acceleration column vector calculation
+        a = (
+            # moon to craft acceleration component
+            mu_moon * r_mc / (np.dot(r_mc, r_mc) ** (3 / 2))
+            # sun to craft acceleration component
+            + mu_sun * r_sc / (np.dot(r_sc, r_sc) ** (3 / 2))
+            # earth to craft acceleration component
+            + mu_earth * r_ec / (np.dot(r_ec, r_ec) ** (3 / 2))
+        )
+
+        return {
+            "vel_x": state.vel_x,
+            "vel_y": state.vel_y,
+            "vel_z": state.vel_z,
+            "acc_x": a[0],
+            "acc_y": a[1],
+            "acc_z": a[2],
```

Comment:
The keys of this dictionary should be the name of the state, and the values should be the _derivatives_ of that state. So it'd be `"x": state.vel_x` and `"vel_x": a[0]`
I know this is kinda confusing, but you're close.

Acceleration isn't actually part of our state vector.